ArduCopter/AP_Arming.cpp

```diff
@@ -352,6 +358,61 @@ bool AP_Arming_Copter::motor_checks(bool display_failure)
     return true;
 }
 
+// check Motor count
+bool AP_Arming_Copter::motor_count_check()
```

Comment:
Add the arming check to inside AP_Motors, then AP_Arming can call the check on motors, which is better as quadplane can benefit from it.